contracts/BondCalculator.sol

```diff
@@ -0,0 +1,301 @@
+            if (numNewUnits >= localParams.targetNewUnits) {
+                discountBooster = uint256(localParams.weightFactors[0]) * 1e18;
+            } else {
+                discountBooster = (uint256(localParams.weightFactors[0]) * numNewUnits * 1e18) /
+                    uint256(localParams.targetNewUnits);
+            }
+        }
+
+        // Second discount booster: booster += k2 * bondVestingTime / productMaxVestingTime
+        // Add vesting time discount booster
+        if (localParams.weightFactors[1] > 0) {
+            if (bondVestingTime == productMaxVestingTime) {
+                discountBooster += uint256(localParams.weightFactors[1]) * 1e18;
+            } else {
+                discountBooster += (uint256(localParams.weightFactors[1]) * bondVestingTime * 1e18) / productMaxVestingTime;
+            }
+        }
+
+        // Third discount booster: booster += k3 * (1 - productPayout(at bonding time) / productSupply)
+        // Add product supply discount booster
+        if (localParams.weightFactors[2] > 0) {
+            if (productPayout == 0) {
+                discountBooster += uint256(localParams.weightFactors[2]) * 1e18;
+            } else {
+                // Get the total product supply
+                productSupply = productSupply + productPayout;
+                discountBooster += uint256(localParams.weightFactors[2]) * (1e18 - ((productPayout * 1e18) / productSupply));
+            }
+        }
+
+        // Fourth discount booster: booster += k4 * getVotes(bonding account) / targetVotingPower
+        // Check the veOLAS balance of a bonding account
+        if (localParams.weightFactors[3] > 0) {
+            uint256 vPower = IVotingEscrow(ve).getVotes(account);
+
+            // If the number of new units exceeds the target, bound by the target number
+            if (vPower >= localParams.targetVotingPower) {
+                discountBooster += uint256(localParams.weightFactors[3]) * 1e18;
+            } else {
+                discountBooster += (uint256(localParams.weightFactors[3]) * vPower * 1e18) /
+                    uint256(localParams.targetVotingPower);
+            }
+        }
+
+        // Normalize discount booster by the max sum of weights
+        discountBooster /= MAX_SUM_WEIGHTS;
+
+        // IDF = 1 + normalized booster
+        idf = 1e18 + discountBooster;
+    }
+
+    /// @dev Calculates the amount of OLAS tokens based on the bonding calculator mechanism accounting for dynamic IDF.
+    /// @param account Account address.
+    /// @param tokenAmount LP token amount.
+    /// @param priceLP LP token price.
+    /// @param bondVestingTime Bond vesting time.
+    /// @param productMaxVestingTime Product max vesting time.
+    /// @param productSupply Current product supply.
+    /// @param productPayout Current product payout.
+    /// @return amountOLAS Resulting amount of OLAS tokens.
+    function calculatePayoutOLAS(
```

Comment:
This is left as is from the `GenericBondCalculator`, which is now obsolete.

Reply:
ok